### DISCUSSION TITLE
livewire:stream

### DISCUSSION THREAD

**Discussion by DirectDev** [livewire/livewire#7047] [7. Bugs (archived)]
### Livewire version

3

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

I have done exactly like documentation : https://livewire.laravel.com/docs/wire-stream.

use Livewire\Component;
 
class CountDown extends Component
{
    public $start = 3;
 
    public function begin()
    {
        while ($start >= 0) {
            // Stream the current count to the browser...
            $this->stream(  
                to: 'count',
                content: $this->start,
                replace: true,
            );
 
            // Pause for 1 second between numbers...
            sleep(1);
 
            // Decrement the counter...
            $this->start = $this->start - 1;
        };
    }
 
    public function render()
    {
        return <<<HTML
        <div>
            <button wire:click="begin">Start count-down</button>
 
            <h1>Count: <span wire:stream="count">{{ $start }}</span></h1> 
        </div>
        HTML;
    }
}

Counter is not streamed.
 I am thinking about missing mandatory server configuration or configuration. 
 Documentation should maybe tell more details.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by bondarde** [comment]
I'm experiencing same issue in #7045. What backend are you using?

**Reply by DirectDev** [reply]
I  have tried both Apache server and Nginx server. PHP 8.2

**Reply by DirectDev** [reply]
I have tried the same script on many different servers. Some of them work, some don't. It's definitly something to see with server configuration.

**Comment by lrljoe** [comment]
If you've got a WAF such as cloudflare, load balancer or similar, make sure it respects the headers that LW uses for this, which include cache control etc

Sharing the response headers from your browser may help here.

**Reply by bondarde** [reply]
I'm having the issue without any WAF or LB, with following response headers:

```
Cache-Control: no-cache, private
Content-Type: text/event-stream;charset=UTF-8
Date: Sun, 15 Oct 2023 17:08:54 GMT
Server: Apache/2.4.52 (Ubuntu)
X-Accel-Buffering: no
X-Livewire-Stream: 1
```

Might PHP-FPM be an issue?

**Comment by anhofmann** [comment]
I see the same behavior: on my local machine (Linux, sail) streaming works fine. On the staging server (Apache, PHP 8.2) the streamed content is only shown when all data was retrieved in the Livewire component.

Here are the response headers, I'm seeing:

```
Cache-Control:
no-cache, private
Content-Type:
text/event-stream;charset=UTF-8
Date:
Tue, 05 Mar 2024 23:04:51 GMT
Server:
Apache
X-Accel-Buffering:
no
X-Livewire-Stream:
1
```

**Reply by joshhanley** [reply]
@anhofmann are you using Apache locally? If not can you do that as a test so we know if it's apache that is not allowing the streaming?

**Comment by bnzo** [comment]
Same behavior when deploying on Laravel vapor. Working fine on local dev. Any lead?

**Reply by Javiersalcedoj** [reply]
Did you find a solution I have the same problem

**Reply by bnzo** [reply]
No, I did not investigate further on this

**Comment by koltejas** [comment]
Any update on this? Its not working as of v3.5, on local and on live server

**Comment by wilianmaique** [comment]
news?

**Reply by anhofmann** [reply]
I've escaped Apache and now use nginx/caddy, so I can't help with this anymore.
